### PR TITLE
tfswitch: init at 0.12.1119

### PR DESCRIPTION
**Pull request by psibi**
###### Motivation for this change

Add tfswitch which allows to easily manage different terraform
versions for different projects.

Tested it locally on a NixOS machine:

```
$ tfswitch --bin=/home/sibi/bin/terraform
Reading required version from terraform file
Reading required version from constraint: ~> 0.13.0
Matched version: 0.13.7
Switched terraform to version "0.13.7"
$ terraform version
Terraform v0.13.7
+ provider registry.terraform.io/hashicorp/aws v2.70.0

Your version of Terraform is out of date! The latest version
is 1.0.1. You can update by downloading from https://www.terraform.io/downloads.html
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -8476,7 +8476,7 @@
     email = "sibi@psibi.in";
     github = "psibi";
     githubId = 737477;
-    name = "Sibi";
+    name = "Sibi Prabakaran";
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:8479]: Please put this into a separate commit.

File: pkgs/applications/networking/cluster/tfswitch/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{ buildGoModule, lib, fetchFromGitHub }:
+buildGoModule rec {
+  pname = "tfswitch";
+  version = "0.12.1119";
+
+  src = fetchFromGitHub {
+    owner = "warrensbox";
+    repo = "terraform-switcher";
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/tfswitch/default.nix:9]:
```suggestion
    rev = version;
```

File: pkgs/applications/networking/cluster/tfswitch/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  runVend = true;
+
+  postInstall = ''
+    # The binary is named tfswitch
+    mv $out/bin/terraform-switcher $out/bin/tfswitch
+  '';
+
+  meta = with lib; {
+    description =
+      "A command line tool to switch between different versions of terraform";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/tfswitch/default.nix:28]:
```suggestion
    description = "A command line tool to switch between different versions of terraform";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (psibi) [reply maintainers/maintainer-list.nix:8479]: Done.

File: pkgs/applications/networking/cluster/tfswitch/default.nix (hunk 3 above)
Comment (psibi) [reply pkgs/applications/networking/cluster/tfswitch/default.nix]: @SuperSandro2000 Addressed this and other comments. Thank you!
